### PR TITLE
Ccw c message

### PR DESCRIPTION
**Pull request by ChihebTrabelsi**
fix https://github.com/Theano/Theano/issues/1680

### CODE REVIEW

**Comment by notoraptor** [comment]
@ChihebTrabelsi , your pull request contains commits that are not related to issue #1680 . You need to rebase your branch `ccw_C_message` with Theano `master` branch ( info for rebasing: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request ).

File: theano/gof/cmodule.py [hunk 1]
```diff
@@ -2290,13 +2290,17 @@ def print_command_line_error():
 
         if status:
             print('===============================')
+            tf = tempfile.NamedTemporaryFile(delete=False)
             for i, l in enumerate(src_code.split('\n')):
                 # gcc put its messages to stderr, so we add ours now
-                print('%05i\t%s' % (i + 1, l), file=sys.stderr)
+                tf.write(str(i + 1) + '\t' + l + '\n')
```
Comment (notoraptor) [inline theano/gof/cmodule.py]: As the C code is now written into a file, you don't need to print line numbers at the begining of lines.

**Comment by ChihebTrabelsi** [comment]
Great, I will remove the line numbers.

File: theano/gof/cmodule.py [hunk 2]
```diff
@@ -2290,13 +2290,17 @@ def print_command_line_error():
 
         if status:
             print('===============================')
+            tf = tempfile.NamedTemporaryFile(delete=False)
             for i, l in enumerate(src_code.split('\n')):
```
Comment (notoraptor) [inline theano/gof/cmodule.py]: Currently, `src_code` is split into lines and each line is written into temporary file. It is faster to remove this `for` loop and write `src_code` entirely in `tf`. Try: `tf.write(src_code)`

**Comment by nouiz** [comment]
I killed travis for rc3 release. we can restart it after

**Comment by nouiz** [comment]
I'm restarting travis

File: theano/gof/cmodule.py [hunk 3]
```diff
@@ -2289,14 +2289,14 @@ def print_command_line_error():
...
-                # gcc put its messages to stderr, so we add ours now
-                print('%05i\t%s' % (i + 1, l), file=sys.stderr)
-            print('===============================')
+            tf = tempfile.NamedTemporaryFile(delete=False)
+            # gcc put its messages to stderr, so we add ours now
+            tf.write(src_code)
+            tf.close()
             print_command_line_error()
             # Print errors just below the command line.
             print(compile_stderr)
```
Comment (nouiz) [inline theano/gof/cmodule.py]: We should put the compilation line (outputed by print_command_line_error()) and the compiler error (compile_stderr) in the error file.

File: theano/gof/cmodule.py [hunk 4]
```diff
@@ -2289,14 +2289,14 @@ def print_command_line_error():
         status = p_out[2]
 
         if status:
-            print('===============================')
-            for i, l in enumerate(src_code.split('\n')):
-                # gcc put its messages to stderr, so we add ours now
-                print('%05i\t%s' % (i + 1, l), file=sys.stderr)
-            print('===============================')
+            tf = tempfile.NamedTemporaryFile(delete=False)
```
Comment (nouiz) [inline theano/gof/cmodule.py]: Can we have a prefix of the file be something like:

theano_compilation_error*

**Comment by nouiz** [comment]
This PR need a rebase.

**Pull request merged**